### PR TITLE
SAMLSettings: implement settings strategy

### PR DESCRIPTION
**Pull request by linoman**
**What is this feature?**

This PR adds a strategy to load SAML settings. 

**Why do we need this feature?**

This strategy is needed to load settings for SAML either from a database source or a configuration file source.

**Who is this feature for?**

IAM

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/identity-access-team/issues/512

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/ssosettings/ssosettings.go [hunk 1]
```diff
@@ -9,10 +9,6 @@ import (
 )
 
 var (
-	// ConfigurableOAuthProviders is a list of OAuth providers that can be configured from the API
-	// TODO: make it configurable
-	ConfigurableOAuthProviders = []string{"github", "gitlab", "google", "generic_oauth", "azuread", "okta"}
-
```
Comment (linoman) [inline pkg/services/ssosettings/ssosettings.go:15]: This code is no longer needed or used anywhere. The list of configurable IdPs can be set at `[sso_settings][configurable_providers]`

File: pkg/services/ssosettings/strategies/saml_strategy.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	cfg              *setting.Cfg
+	settingsProvider setting.Provider
+}
+
+var _ ssosettings.FallbackStrategy = (*SAMLStrategy)(nil)
+
+func NewSAMLStrategy(cfg *setting.Cfg, settingsProvider setting.Provider) *SAMLStrategy {
+	return &SAMLStrategy{
+		cfg:              cfg,
+		settingsProvider: settingsProvider,
```
[the remark this answers is not in the document]
Comment (linoman) [reply pkg/services/ssosettings/strategies/saml_strategy.go:21]: There's a difference between the OAuth provider and the saml provider. The saml provider will load the settings from the SettingsProvider but not from the configuration. By querying from the SettingsProvider, we're ensuring to have the latest version, a reloading of setting without the need of a timed refresh (like for OAuth) or a instance restart.

However, now that you point this out, `cfg` is not used at all and can be removed.

File: pkg/services/ssosettings/strategies/saml_strategy.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+func (s *SAMLStrategy) IsMatch(provider string) bool {
+	return provider == "saml"
+}
+
+func (s *SAMLStrategy) GetProviderConfig(_ context.Context, provider string) (map[string]any, error) {
+	return s.loadSAMLSettings(), nil
+}
+
+func (s *SAMLStrategy) loadSAMLSettings() map[string]any {
+	section := s.settingsProvider.Section("auth.saml")
```
[the remark this answers is not in the document]
Comment (linoman) [reply pkg/services/ssosettings/strategies/saml_strategy.go:34]: My previous comment addresses the difference of loading files from the 2 type of providers.

SAML configurations are stored in the same `custom.ini` file as we use in Grafana OSS.  You can take a look at it in the Grafana Enterprise repo.

https://github.com/grafana/grafana-enterprise/blob/4598b32c8d14f2b9fe05ac840c498419150466b3/src/settings/defaults.ini#L146

File: pkg/services/ssosettings/strategies/saml_strategy.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (linoman) [reply pkg/services/ssosettings/strategies/saml_strategy.go]: Removed!

**Review by dmihai — APPROVED** [review]

**Pull request merged**